test/unit/search/search_configuration_test.cpp

```diff
@@ -54,131 +59,42 @@ TYPED_TEST(search_configuration_test, configuration_exists)
 
```

Comment:
You removed the tests that checked whether the correct values are set when you use values other than 0 or default ctor.
Is this intended?

Also IIRC, we are throwing if the config is invalid? E.g. more substitution errors than total errors, etc.
Is this tested somewhere?

Reply:
The deleted tests tested whether the configuration elements are filled automatically if only parts of the config are set.
Due to the new structure, the tests can no longer be performed. (You cannot compare the piped elements.)
The tests are tested indirectly in the search_test by the application of the config.